torch/csrc/distributed/rpc/utils.cpp

```diff
@@ -353,6 +353,105 @@ std::pair<std::vector<char>, std::vector<at::Tensor>> wireDeserialize(
   return {std::move(payload), std::move(tensors)};
 }
 
+TensorPipeEntry tensorpipeSerialize(const Message& rpcMessage) {
+  tensorpipe::Message tpMessage;
+  std::vector<torch::Tensor> reservedTensors;
+  std::vector<std::vector<uint8_t>> copiedTensors;
+
+  const std::vector<char>& payload = rpcMessage.payload();
+  c10::List<at::Tensor> tensors = cloneSparseTensors(rpcMessage.tensors());
+
+  // Payload
+  tpMessage.data = (uint8_t*)(payload.data());
```

Comment:
We should add a note to `tensorpipeSerialize` docs that the caller needs to keep rpcMessage alive for the returned TensorPipeEntry to be valid since TensorPipeEntry has references to the original rpcMessage.